metrics/sinks/gcm/gcm.go

```diff
@@ -321,21 +325,43 @@ func CreateGCMSink(uri *url.URL) (core.DataSink, error) {
 		return nil, fmt.Errorf("invalid metrics parameter: %s", metrics)
 	}
 
-	if err := gce_util.EnsureOnGCE(); err != nil {
-		return nil, err
-	}
-
-	// Detect project ID
-	projectId, err := gce.ProjectID()
+	client, err := google.DefaultClient(oauth2.NoContext, gcm.MonitoringScope)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating oauth2 client: %s", err)
 	}
 
-	// Create Google Cloud Monitoring service.
-	client := oauth2.NewClient(oauth2.NoContext, google.ComputeTokenSource(""))
 	gcmService, err := gcm.New(client)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating GCM service: %s", err)
+	}
+
+	// Try and get the project ID from an environment variable first
+	var projectId string
+	projectId = os.Getenv(gcpProjectId)
+	if projectId == "" {
+		// If the variable is not set, move on to the default credentials file
+		file := os.Getenv(gcpCredentialEnv)
```

Comment:
[Optional] The Google OAuth2 libraries also look for this file in the default location `/etc/google/auth/application_default_credentials.json`. @crassirostris and @piosz should decide if it makes sense to match this behavior.